core/src/main/java/io/kestra/core/runners/WorkerTriggerRealtimeThread.java

```diff
@@ -0,0 +1,52 @@
+package io.kestra.core.runners;
+
+import io.kestra.core.models.executions.Execution;
+import io.kestra.core.models.triggers.RealtimeTriggerInterface;
+import org.reactivestreams.Publisher;
+import reactor.core.publisher.Flux;
+
+import java.util.function.Consumer;
+
+import static io.kestra.core.models.flows.State.Type.SUCCESS;
+
+public class WorkerTriggerRealtimeThread extends AbstractWorkerTriggerThread {
+    RealtimeTriggerInterface streamingTrigger;
+    Consumer<? super Throwable> onError;
+    Consumer<Execution> onNext;
+
+    public WorkerTriggerRealtimeThread(
+        WorkerTrigger workerTrigger,
+        RealtimeTriggerInterface realtimeTrigger,
+        Consumer<? super Throwable> onError,
+        Consumer<Execution> onNext
+    ) {
+        super(workerTrigger.getConditionContext().getRunContext(), realtimeTrigger.getClass().getName(), workerTrigger);
+        this.streamingTrigger = realtimeTrigger;
+        this.onError = onError;
+        this.onNext = onNext;
+    }
+
+    @Override
+    public void run() {
+        Thread.currentThread().setContextClassLoader(this.streamingTrigger.getClass().getClassLoader());
+
+        Publisher<Execution> evaluate;
+        try {
+            evaluate = streamingTrigger.evaluate(
+                workerTrigger.getConditionContext().withRunContext(runContext),
+                workerTrigger.getTriggerContext()
+            );
+            taskState = SUCCESS;
```

Comment:
Should we plan a realtime trigger to end at some point?
Otherwise, this is useless.

Reply:
Only keep in case a RealtimeTrigger will stop (for edge case), to be restarted automatically